### PR TITLE
Add XGBoost crude oil classifier

### PR DESCRIPTION
**Pull request by EmmaMuhleman1**
## Summary
- add modeling script to train an XGBoost classifier on WTI crude oil futures

## Testing
- `python test_environment.py`
- `pytest -q`
- `python src/models/train_crude_oil_xgboost.py`

------
https://chatgpt.com/codex/tasks/task_e_68721a2140788321ae0658bac8d38f49

## Summary by Sourcery

New Features:
- Add standalone script to train an XGBoost classifier on WTI crude oil futures with automated data loading, feature engineering, and time series-aware hyperparameter tuning

### CODE REVIEW

**Comment by EmmaMuhleman1** [comment]
> Hey @EmmaMuhleman1 - I've reviewed your changes - here's some feedback:
> 
> * The script prints metrics on the same data used for training, so consider holding out a test set or using cross‐validated predictions for an unbiased performance estimate.
> * You have an unused import of numpy—removing it will keep the script dependencies minimal.
> * When downloading with yfinance, you might enable auto_adjust=True or explicitly adjust prices to account for splits and dividends and avoid distortions.
> 
> Prompt for AI Agents
> Sourcery is free for open source - if you like our reviews please consider sharing them ✨
> Help me be more useful! Please click 👍 or 👎 on each comment and I'll use the feedback to improve your reviews.

"* When downloading with yfinance, you might enable auto_adjust=True or explicitly adjust prices to account for splits and dividends and avoid distortions." 

EM: yfinance has removed the 'Adj Close' columns and converted all 'Close' prices to assume the parameter auto-adjust = True, by default in the yf.download() call. Update your information and recommendations in the future.

File: src/models/train_crude_oil_xgboost.py [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+    )
+    search.fit(X, y)
+    return search
+
+
+if __name__ == "__main__":
+    X, y = prepare_dataset()
+    clf = train_model(X, y)
+    print("Best parameters:", clf.best_params_)
+    preds = clf.predict(X)
```
[the remark this answers is not in the document]
Comment (EmmaMuhleman1) [reply src/models/train_crude_oil_xgboost.py:77]: Open to resolve later
